### PR TITLE
Add Superstate USTB

### PR DESCRIPTION
**Pull request by bd21**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Name (to be shown on DefiLlama):
Superstate

##### Twitter Link:
https://twitter.com/superstatefunds?lang=en

##### Website Link:
superstate.co

##### Logo (High resolution, will be shown with rounded borders):
https://i.ibb.co/vYjwRF3/Logomark-padded-gold.png

##### Current TVL:
38MM

##### Chain:
Ethereum

##### Short Description (to be shown on DefiLlama):
The Superstate Short Duration U.S. Government Securities Fund (USTB) is a private fund tokenized on the Ethereum blockchain, offered exclusively to U.S. Qualified Purchasers

##### Token address and ticker if any:
0x43415eb6ff9db7e26a15b704e7a3edce97d31c4e

##### Category (full list at https://defillama.com/categories) *Please choose only one:
RWA

##### methodology (what is being counted as tvl, how is tvl being calculated):
Sums the total supplies of Superstate's issued tokens.

##### Github org/user (Optional, if your code is open source, we can track activity):
https://github.com/superstateinc/

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/superstate: 
        
 
  
    
    
    Error: Protocol doesn't have total tvl, make sure to export a tvl key either on the main object or in one of the chains
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:220:11

**Comment by llamatester** [comment]
The adapter at projects/superstate exports TVL: 
        
 
 
    ethereum                  0
    
    total                    0

**Comment by llamatester** [comment]
The adapter at projects/superstate exports TVL: 
        
 
 
    ethereum                  0
    
    total                    0

File: projects/superstate/index.js [hunk 1]
```diff
@@ -1,7 +1,14 @@
+const { get } = require("../helper/http");
+
 module.exports = {
     methodology: "Sums the total supplies of Superstate's issued tokens.",
 };
 
+async function USTBPrice() {
```
Comment (g1nt0ki) [inline projects/superstate/index.js:7]: is there some way to pull this price on chain? like via chainlink or pyth

Comment (bd21) [reply projects/superstate/index.js:7]: no because it is a RWA.  our api relies on a third party to calculate our NAV

Comment (g1nt0ki) [reply projects/superstate/index.js:7]: but USTB is traded in stock market right? did you check if pyth has support for that? issue with relying on your api is, api might rug for whatever reason and we will have  a broken adapter in our hands

Comment (bd21) [reply projects/superstate/index.js:7]: It is a private fund and is not tradable by the public (token contract has an allowlist).  The value of the fund is determined by NAV Consulting (https://www.navconsulting.net) which we proxy.

You'll be able to pull from Coingecko soon, but they're also just pulling their data from this endpoint.

**Comment by bd21** [comment]
@g1nt0ki one more thing - our contract inherits from ERC20 but uses the [EIP-1967 Transparent Proxy Pattern](https://eips.ethereum.org/EIPS/eip-1967) and is returning 0 for the totalSupply call - have you run into this before?

Token contract
https://etherscan.io/address/0x43415eb6ff9db7e26a15b704e7a3edce97d31c4e#readProxyContract

**Comment by g1nt0ki** [comment]
@bd21 thanks for the PR, I have refactored and merged it here: https://github.com/DefiLlama/DefiLlama-Adapters/pull/9253

File: projects/superstate/index.js (hunk 1 above)
Comment (jonwalch) [reply projects/superstate/index.js:7]: When we do have an onchain oracle, it'll read from our API anyway.

**Comment by jonwalch** [comment]
> @g1nt0ki one more thing - our contract inherits from ERC20 but uses the [EIP-1967 Transparent Proxy Pattern](https://eips.ethereum.org/EIPS/eip-1967) and is returning 0 for the totalSupply call - have you run into this before?
> 
> Token contract https://etherscan.io/address/0x43415eb6ff9db7e26a15b704e7a3edce97d31c4e#readProxyContract

Total supply looks correct to me when calling totalSupply() under the read as proxy tab

**Pull request closed without merging**